src/libos/crates/async-rt/src/scheduler/local_scheduler/mod.rs

```diff
@@ -113,19 +113,26 @@ impl<E: SchedEntity> LocalScheduler<E> {
             return;
         }
 
+        let is_yielded = sched_state.set_yielded(false);
+
+        let last_vcpu = sched_state.vcpu();
         sched_state.set_vcpu(self.this_vcpu);
 
         // Depending on whether the entity still has remaining timeslice,
         // enqueue it into the front or back runqueues.
         let mut rqs = self.lock_rqs();
-        if sched_state.timeslice() > 0 {
+        if last_vcpu.is_some() && !is_yielded {
             let front_rqs = rqs.front_mut();
-            front_rqs.enqueue(entity.clone());
+            front_rqs.enqueue_jump(entity.clone());
```

Comment:
Why not reassgin time slices if there are no remaining time slices?

Reply:
Our scheduler is not preemptive, so time slice is just a record for running time. However, those queue-jump tasks will be poped out immediately, assigning time slice may be more useful for next enqueueing rather than this time.

Reply:
Thanks for your explanation